src/utils/CometAdapter.cpp

```diff
@@ -0,0 +1,589 @@
+//Doxygen docu
+//-------------------------------------------------------------
+
+/**
+    @page TOPP_CometAdapter CometAdapter
+
+    @brief Identifies peptides in MS/MS spectra via Comet.
+
+<CENTER>
+    <table>
+        <tr>
+            <td ALIGN = "center" BGCOLOR="#EBEBEB"> pot. predecessor tools </td>
+            <td VALIGN="middle" ROWSPAN=2> \f$ \longrightarrow \f$ CometAdapter \f$ \longrightarrow \f$</td>
+            <td ALIGN = "center" BGCOLOR="#EBEBEB"> pot. successor tools </td>
+        </tr>
+        <tr>
+            <td VALIGN="middle" ALIGN = "center" ROWSPAN=1> any signal-/preprocessing tool @n (in mzML format)</td>
+            <td VALIGN="middle" ALIGN = "center" ROWSPAN=1> @ref TOPP_IDFilter or @n any protein/peptide processing tool</td>
+        </tr>
+    </table>
+</CENTER>
+
+    @em Comet must be installed before this wrapper can be used. This wrapper
+    has been successfully tested with version 2016.01.2 of Comet.
+
+    Comet settings not exposed by this adapter can be directly adjusted using a param file, which can be generated using comet -p.
+    By default, All (!) parameters available explicitly via this param file will take precedence over the wrapper parameters.
+
+    Parameter names have been changed to match names found in other search engine adapters, however some are Comet specific.
+    For a detailed description of all available parameters check the Comet documentation at http://comet-ms.sourceforge.net/parameters/parameters_201601/
+
+    <B>The command line parameters of this tool are:</B>
+    @verbinclude TOPP_CometAdapter.cli
+    <B>INI file documentation of this tool:</B>
+    @htmlinclude TOPP_CometAdapter.html
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
+
+
+class TOPPCometAdapter :
+  public TOPPBase
+{
+public:
+  TOPPCometAdapter() :
+    TOPPBase("CometAdapter", "Annotates MS/MS spectra using Comet.")
+  {
+  }
+
+protected:
+  void registerOptionsAndFlags_()
+  {
+
+    registerInputFile_("in", "<file>", "", "Input file");
+    setValidFormats_("in", ListUtils::create<String>("mzML"));
+    registerOutputFile_("out", "<file>", "", "Output file");
+    setValidFormats_("out", ListUtils::create<String>("idXML"));
+    registerInputFile_("database", "<file>", "", "FASTA file", true, false, ListUtils::create<String>("skipexists"));
+    setValidFormats_("database", ListUtils::create<String>("FASTA"));
+    registerInputFile_("comet_executable", "<executable>",
+      // choose the default value according to the platform where it will be executed
+      "comet.exe",
+      "Comet executable of the installation e.g. 'comet.exe'", true, false, ListUtils::create<String>("skipexists"));
+
+    //
+    // Optional parameters //
+    //
+
+    //Files
+    registerOutputFile_("pin_out", "<file>", "", "Output file - for Percolator input", false);
+    setValidFormats_("pin_out", ListUtils::create<String>("pin"));
```

Comment:
I think this would be okay as long as Comet accepts this file type and does not rename it while it writes the file (e.g. adds ".pin" automatically). I would vote for allowing both file types (csv as first, i.e. standard), what do you think?